### PR TITLE
feat: Add confidence score support to transcription output

### PR DESCRIPTION
**Pull request by samasante**
Adds confidence scores (0.0-1.0) to the transcription output using entropy from the model's logits.

Implemented across all three decoders (TDT, RNNT, CTC) and shows up in both verbose mode and JSON output. Uses the standard `1 - (H / H_max)` approach - no extra inference needed.

Tested with real audio and the scores correlate well with audio quality (clear speech gets 95-99%, noisy/overlapping audio drops to 84-92%).

### CODE REVIEW

File: parakeet_mlx/alignment.py [hunk 1]
```diff
@@ -20,12 +21,18 @@ class AlignedSentence:
...
     def __post_init__(self) -> None:
         self.tokens = list(sorted(self.tokens, key=lambda x: x.start))
         self.start = self.tokens[0].start
         self.end = self.tokens[-1].end
         self.duration = self.end - self.start
+        # Compute geometric mean of token confidences
+        if self.tokens:
+            import mlx.core as mx
+            confidences = mx.array([t.confidence for t in self.tokens])
+            self.confidence = float(mx.exp(mx.mean(mx.log(confidences + 1e-10))))
```
Comment (senstella) [inline parakeet_mlx/alignment.py:35]: We can assume that tokens are not empty list. (Already assuming in line 27!) Also, MLX can be pretty slow at this immediate calculation since it [builds graph](https://gist.github.com/awni/4beb1f7dfefc6f9426f3a7deee74af50) first. I think we can use NumPy here! Also I think import can be placed on top of the file.

```suggestion
        confidences = np.array([t.confidence for t in self.tokens])
        self.confidence = float(np.exp(np.mean(np.log(confidences + 1e-10))))
```

**Review by senstella — APPROVED** [review]
Hello! This is a really awesome feature to have in the library. Thanks for working on it! I've commented on a very minor style suggestion. It'd be nice if you could take a quick look!

**Comment by senstella** [comment]
Looks good to me! I merged the PR. Thank you for your contribution, I really appreciate it!
